lib/route.js

```diff
@@ -307,7 +307,7 @@ function buildRouting (options) {
 
     req.id = req.headers[requestIdHeader] || genReqId(req)
     req.originalUrl = req.url
-    var hostname = req.headers.host
+    var hostname = req.headers[':authority'] || req.headers.host
```

Comment:
Would you mind to invert those?

```suggestion
    var hostname = req.headers.host || req.headers[':authority']
```